### PR TITLE
refactor(bottom-navigation-item): optimize code

### PR DESCRIPTION
**Pull request by chouchouji**
## Checklist

List of tasks you have already done and plan to do.

## Change information

Describe your modifications here.

## Issues

The issues you want to close, formatted as close #1.

## Related Links

Links related to this pr.

### CODE REVIEW

File: packages/varlet-ui/src/bottom-navigation-item/BottomNavigationItem.vue [hunk 1]
```diff
@@ -53,7 +44,7 @@ export default defineComponent({
   props,
   setup(props) {
     const name = computed<string | undefined>(() => props.name)
-    const badge = computed<boolean | BadgeProps>(() => props.badge)
+    const isActive = computed<boolean>(() => active.value === name.value || active.value === index.value)
```
Comment (haoziqaq) [inline packages/varlet-ui/src/bottom-navigation-item/BottomNavigationItem.vue:47]: Don't use conditional branch statements in computed. It is easy to create unnecessary mental burden due to dependence on collection issues.

Comment (chouchouji) [reply packages/varlet-ui/src/bottom-navigation-item/BottomNavigationItem.vue:47]: sorry, I forget it. It seems that using `includes` is a better choice.

Comment (chouchouji) [reply packages/varlet-ui/src/bottom-navigation-item/BottomNavigationItem.vue]: done

File: packages/varlet-ui/src/bottom-navigation-item/BottomNavigationItem.vue [hunk 2]
```diff
@@ -65,33 +56,27 @@ export default defineComponent({
...
 
     watch(
-      () => badge.value,
+      () => props.badge,
       (newValue) => {
-        badgeProps.value = newValue === true ? defaultBadgeProps : badge.value
+        badgeProps.value = newValue === true ? defaultBadgeProps : props.badge
       },
       { immediate: true }
     )
```
Comment (zhangmo8) [inline packages/varlet-ui/src/bottom-navigation-item/BottomNavigationItem.vue:64]: Why not got the badgeProps.value in computed

Comment (chouchouji) [reply packages/varlet-ui/src/bottom-navigation-item/BottomNavigationItem.vue:64]: Good idea,  I will replace `watch` with `computed`

**Pull request merged**